src/odemis/gui/comp/miccanvas.py

```diff
@@ -421,28 +425,58 @@ def _convert_streams_to_images(self):
         # add the images in order
         ims = []
         im_cache = {}
-        for rgbim, blend_mode, name in images:
-            # Get converted RGBA image from cache, or create it and cache it
-            # On large images it costs 100 ms (per image and per canvas)
-            im_id = id(rgbim)
-            if im_id in self.images_cache:
-                rgba_im = self.images_cache[im_id]
+        tiles_cache = {}
+        for rgbim, blend_mode, name, stream in images:
+            if isinstance(rgbim, tuple): # tuple of tuple of tiles
+                if len(rgbim) == 0 or len(rgbim[0]) == 0:
+                    continue
+                first_tile = rgbim[0][0]
+                md = first_tile.metadata
+                new_array = []
+                for tile_column in rgbim:
+                    new_array_col = []
+                    for tile in tile_column:
+                        tile_id = id(tile)
+                        if tile_id in self.tiles_cache:
+                            rgba_tile = self.tiles_cache[tile_id]
```

Comment:
No need to have two dicts, `tile_cache` and `images_cache`. Each id() should be different, so, instead, use `images_cache` in both cases.